lib/repository/repository.dart

```diff
@@ -112,14 +115,59 @@ abstract class Repository {
   /// Input: [id] the id of the nft which the user wants to delete
   /// Output: [bool] returns whether the operation is successful or not
   Future<Either<Failure, bool>> deleteNft(int id);
+
+  /// This function picks a file with the given format from device storage
+  /// Input: [format] it is the file format which needs to be picked from local storage
+  /// returns [PickedFileModel] the selected file or [Failure] if aborted
+  Future<Either<Failure, PickedFileModel>> pickFile(NftFormat format);
+
+  /// This function takes the file and returns a compressed version of that file
+  /// Input: [file] it takes the file that needs to be compressed
+  /// Output: [File] returns the compressed file
+  Future<Either<Failure, File?>> compressAndGetFile(File file);
```

Comment:
It should be a file if there is no file throw an error. 

Output documentation is not according to our standards.